u_fingerprinter.go

```diff
@@ -277,7 +277,7 @@ func (f *Fingerprinter) FingerprintClientHello(data []byte) (*ClientHelloSpec, e
 				ks.Group = CurveID(unGREASEUint16(group))
 				// if not GREASE, key share data will be discarded as it should
 				// be generated per connection
-				if ks.Group != GREASE_PLACEHOLDER {
+				if !isGREASEUint16(group) {
```

Comment:
Is there a specific reason for this change? Since we have already saved the `unGREASEUint16(group)` to `ks.Group`, checking if `ks.Group == GREASE_PLACEHOLDER` is equivalent to checking `isGREASEUint16(group)` if no bug.